### ISSUE TITLE
Maintain editor tab order and active tab when restoring Data Explorer editors

### ISSUE THREAD

**Issue by sharon-wang** [posit-dev/positron#5489]
## Describe the issue:

#5475 implemented the restoring of Data Explorer tabs upon reloading the IDE for #3172. However, the Data Explorer tabs are restored after any code editors are restored and tab order is not tracked, which can cause the editor tab ordering to change.

## Steps to reproduce the issue:

1. Start an interpreter, e.g. `R 4.4.0`
2. View at least one data frame in the Data Explorer, e.g. `View(iris)`
3. Open a code file and organize the editor tabs such that the Data Explorer tabs are not all at the end
    - e.g., ![Image](https://github.com/user-attachments/assets/e11b4231-2567-4476-9ecf-ad9aef02b6fb)
4. Reload the IDE, e.g. `Developer: Reload Window` command in the Command Prompt (See https://github.com/posit-dev/positron/issues/5462 for more ways to reload)
5. See that the Data Explorer tab(s) have been restored to the end of the tabs
    - e.g., ![Image](https://github.com/user-attachments/assets/d3286939-b813-41a4-803b-1c323e149c7a)

## Expected or desired behavior:

The tab order before reloading the IDE should be preserved, and a reopened Data Explorer tab should not become the active tab if another editor were previously the active tab.

From https://github.com/posit-dev/positron/pull/5475#pullrequestreview-2456124881

> some changes in the data explorer service itself (to mark these tabs as custom editors that can be restored?) are needed for a better UX.

## Were there any error messages in the UI, Output panel, or Developer Tools console?

N/A
